<?xml version="1.0" encoding="UTF-8"?>
<!-- 
    This is the production launch file to be used with the real roboclaw device.
-->
 <launch>

    <node name="roboclaw" pkg="roboclaw_driver" type="roboclaw_node.py">
        <param name="test_mode" type="bool" value="false"/>
        <param name="dev_name" type="str" value="/dev/ttyACM0"/>
        <param name="baud" type="int" value="115200"/>
        <param name="address" type="int" value="128"/>
        <param name="loop_hz" type="int" value="100"/>
        <param name="deadman_secs" type="int" value="3"/>
        <param name="speed_cmd_topic" type="str" value="roboclaw/speed_command"/>
        <param name="stats_topic" type="str" value="roboclaw/stats"/>
    </node>

    <node name="motor_control" pkg="roboclaw_driver" type="motor_control_node.py">
        <param name="robot_cmd_topic" type="str" value="control_vel_cmd"/>
        <!-- We can add an implementation of 4WD if needed -->
        <param name="motor_config" type="str" value="2WD"/> 
        <!-- Width between the wheels in mm-->
        <param name="base_width" type="int" value="200"/> 
        <!-- The number of encoder ticks per meter of movement straight forwards-->
        <param name="ticks_per_meter" type="double" value="4342.2"/> 
        <!-- Maximum time before the motor automatically stops after a command in seconds-->
        <param name="max_secs" type="int" value="1000"/> 
    </node>
    <!-- Front wheels driver. Edit connection details
    <node name="roboclaw" pkg="roboclaw_driver" type="roboclaw_node.py">
        <param name="test_mode" type="bool" value="false"/>
        <param name="dev_name" type="str" value="/dev/ttyACM0"/>
        <param name="baud" type="int" value="115200"/>
        <param name="address" type="int" value="128"/>
        <param name="loop_hz" type="int" value="100"/>
        <param name="deadman_secs" type="int" value="3"/>
        <param name="speed_cmd_topic" type="str" value="roboclaw/speed_command/front"/>
        <param name="stats_topic" type="str" value="roboclaw/stats"/>
    </node>-->


    <!-- Back wheels driver. Edit connection details
    <node name="roboclaw" pkg="roboclaw_driver" type="roboclaw_node.py">
        <param name="test_mode" type="bool" value="false"/>
        <param name="dev_name" type="str" value="/dev/ttyACM0"/>
        <param name="baud" type="int" value="115200"/>
        <param name="address" type="int" value="128"/>
        <param name="loop_hz" type="int" value="100"/>
        <param name="deadman_secs" type="int" value="3"/>
        <param name="speed_cmd_topic" type="str" value="roboclaw/speed_command"/>
        <param name="stats_topic" type="str" value="roboclaw/stats"/>
    </node> -->

</launch>